noir_stdlib/src/meta/mod.nr

```diff
@@ -41,3 +41,45 @@ pub comptime fn derive(s: StructDefinition, traits: [TraitDefinition]) -> Quoted
 unconstrained pub comptime fn derive_via(t: TraitDefinition, f: DeriveFunction) {
     HANDLERS.insert(t, f);
 }
+
+/// `make_impl` is a helper function to make a simple impl, usually while deriving a trait.
+/// This impl has a couple assumptions:
+/// 1. The impl only has one function, with the signature `function_signature`
+/// 2. The trait itself does not have any generics.
+///
+/// While these assumptions are met, `make_impl` will create an impl from a StructDefinition,
+/// automatically filling in the required generics from the struct, along with the where clause.
+/// The function body is created by mapping each field with `for_each_field` and joining the
+/// results with `join_fields_with`. The result of this is passed to the `body` function for
+/// any final processing - e.g. wrapping each field in a `StructConstructor { .. }` expression.
+///
+/// See `derive_eq` and `derive_default` for example usage.
+pub comptime fn make_impl<Env1, Env2>(
```

Comment:
Updated. I chose `make_struct_impl` since I thought `make_simple_struct_impl` was getting a bit into the weeds but was also somewhat confusing in what "simple" meant. I think we can change the name of this function later if we need to.